### DISCUSSION TITLE
PDF tab preview not working

### DISCUSSION THREAD

**Discussion by Krrost** [coder/code-server#6062] [General]
Hi,

I have [vscode-server](https://hub.docker.com/r/linuxserver/code-server) in a docker container, hosted on my domain through  Traefik reverse proxy, with LatexWorkshop and Texlive installed. 

Everything seems to be ok, but I can't see the pdf preview in the tab, there is the pdf viewer, but nothing inside, [here's a screenshot](https://i.stack.imgur.com/14QJ0.png). If I go to get the file, the pdf is generated normally. 

LatexWorkshop log:

```
[Commander] VIEW command invoked with mode: tab.
[Manager] Current workspace folders: ["file://%WS1%","file://%WS2%"]
[Manager] Found root file from active editor: %WS2%/aaa.tex
[Manager] Keep using the same root file: %WS2%/aaa.tex
[Structure] Structure updated with 0 root sections for %WS2%/aaa.tex .
[Viewer][Panel] Internal PDF viewer at https://mydomain.com/proxy/40451/viewer.html?file=pdf..xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx .
[Cacher][Watcher] Watched %WS2%/aaa.pdf .
[Viewer] Open PDF tab for file://%WS2%/aaa.pdf
[Viewer] Handle data type: open
```
The pdf is accessible from the address `https://mydomain.com/proxy/40451/.....`, it correctly shows the [pdf in a browser tab](https://i.stack.imgur.com/cwJnY.png). So the "View in web browser" option works.

Devtools log:
![image](https://user-images.githubusercontent.com/81440126/223982586-30d97add-56bc-47e1-9ae7-00cfaab0b628.png)

Any ideas? Thanks.
Thanks

**Comment by code-asher** [comment]
I reproduced on code-server v4.10.1.  It works on v4.8.3.

It looks like the port is missing from the URLs the plugin is trying to load.

The plugin uses relative URLs like `build/pdf.worker.js` and the iframe URL itself is `https://<domain>/proxy/37127/viewer.html?file=pdf..<some long string>` so it seems like that should resolve properly as `https://<domain>/proxy/37127/build/pdf.worker.js` but somehow it seems the port gets dropped.

Like you mentioned though it works fine outside the iframe so...right now my guess is that the service worker that intercepts requests VS Code sticks in between the main thread and web views is doing something wrong but this will need further investigation.

**Reply by Krrost** [reply]
> I reproduced on code-server v4.10.1. It works on v4.8.3.
> 
> It looks like the port is missing from the URLs the plugin is trying to load.
> 
> The plugin uses relative URLs like `build/pdf.worker.js` and the iframe URL itself is `https://<domain>/proxy/37127/viewer.html?file=pdf..<some long string>` so it seems like that should resolve properly as `https://<domain>/proxy/37127/build/pdf.worker.js` but somehow it seems the port gets dropped.
> 
> Like you mentioned though it works fine outside the iframe so...right now my guess is that the service worker that intercepts requests VS Code sticks in between the main thread and web views is doing something wrong but this will need further investigation.

Thanks for the reply.

I pulled v4.8.3 but It's still not working.
I also tried to redirect through Traefik:
```
 - "traefik.http.middlewares.pdf-rewrite.replacepathregex.regex=^/proxy/build/pdf.worker.js"
 - "traefik.http.middlewares.pdf-rewrite.replacepathregex.replacement=/proxy/33909/build/pdf.worker.js"
 ```
 
 But still no luck.

**Reply by code-asher** [reply]
We just released 4.11.0 and that one is working for me.  If 4.8.3 did not work for you though maybe this one will not work either.  Check to see if you get different errors!
